nesc/whip6/platforms/parts/mcu/cc26xx/uart/HalUARTNonBlockingWritePrv.nc

```diff
@@ -0,0 +1,51 @@
+/*
+ * whip6: Warsaw High-performance IPv6.
+ *
+ * Copyright (c) 2017 University of Warsaw
+ * All rights reserved.
+ *
+ * This file is distributed under the terms in the attached LICENSE
+ * files.
+ */
+#include <stdio.h>
+#include "scif_uart_emulator.h"
+
+module HalUARTNonBlockingWritePrv {
+    provides interface NonBlockingWrite<uint8_t>;
+} implementation {
+    uint32_t m_lost_bytes = 0;
+    char m_last_char = 0;;
+    char m_buffer[16];
+
+    async command error_t NonBlockingWrite.write(uint8_t value) {
+        uint32_t free = 2 * (SCIF_UART_TX_FIFO_MAX_COUNT - scifUartGetTxFifoCount());
+        if (m_lost_bytes > 0) {
+            uint32_t needed;
+            sprintf(m_buffer, "\nLOST:%d\n%c", m_lost_bytes, value);
+            needed = strlen(m_buffer);
```

Comment:
But no, it seems that this sprintf has a different signature than the one from cstdio.